indexer/ai/src/main/java/com/google/cloudsearch/ai/BaseAISkill.java

```diff
@@ -0,0 +1,133 @@
+package com.google.cloudsearch.ai;
+
+import com.google.cloudsearch.exceptions.InvalidConfigException;
+import org.json.simple.JSONArray;
+import org.json.simple.JSONObject;
+import java.util.*;
+
+/**
+ * Abstract class BaseAISkill implements the common functionality required for the different skills.
+ */
+public abstract class BaseAISkill implements AISkill {
+
+    private String aiSkillName = "";
+    private List<OutputMapping> outputMappings = new ArrayList<OutputMapping>();
+    private Map<String, Boolean> schemaInfo = new HashMap<String, Boolean>();
+    private Boolean isSchemaInfoStored = false;
+
+    /**
+     * Parses the CloudsSearch Schema and stores the object and property names
+     * @param schema    CloudSearch schema
+     */
+    private void storeSchemaInfo(JSONObject schema){
+        JSONArray schemaObjects = (JSONArray) schema.get("objectDefinitions");
+        for(Object schemaObj : schemaObjects){
```

Comment:
how would this handle properties which has subobjects ?

Reply:
Currently both multiple objects and subobjects are not supported.